Default more constructors / destructors in WebKit

cdumez opened the pull request:
#### 8106797ab44ec9bcbb004abe546b122fefaccbd4
<pre>
Default more constructors / destructors in WebKit
<a href="https://bugs.webkit.org/show_bug.cgi?id=300506">https://bugs.webkit.org/show_bug.cgi?id=300506</a>

Reviewed by Sam Weinig and Darin Adler.

Default more constructors / destructors in WebKit.

* Source/JavaScriptCore/bytecode/Watchpoint.h:
* Source/WTF/wtf/CheckedRef.h:
* Source/WTF/wtf/ConcurrentVector.h:
* Source/WTF/wtf/MallocCommon.h:
(WTF::ForbidMallocUseForCurrentThreadScope::~ForbidMallocUseForCurrentThreadScope):
* Source/WTF/wtf/MetaAllocator.h:
* Source/WTF/wtf/PreciseSum.h:
* Source/WTF/wtf/RefTrackerMixin.h:
* Source/WTF/wtf/WeakPtrImpl.h:
* Source/WebCore/Modules/applepay/ApplePayError.h:
* Source/WebCore/Modules/gamepad/GamepadEvent.h:
* Source/WebCore/Modules/mediacontrols/MediaControlsUtils.h:
(WebCore::MediaControlsUtils::create):
* Source/WebCore/Modules/speech/SpeechRecognitionCaptureSource.h:
* Source/WebCore/Modules/streams/ReadableStreamBYOBRequest.h:
* Source/WebCore/Modules/streams/ReadableStreamReadRequest.cpp:
(WebCore::ReadableStreamDefaultReadRequest::create):
(WebCore::ReadableStreamDefaultReadIntoRequest::create):
* Source/WebCore/Modules/streams/StreamPipeToUtilities.cpp:
(WebCore::PipeToDefaultReadRequest::create):
* Source/WebCore/Modules/streams/StreamTeeUtilities.cpp:
(WebCore::TeeDefaultReadRequest::create):
(WebCore::TeeBYOBReadRequest::create):
* Source/WebCore/Modules/webcodecs/WebCodecsEncodedAudioChunk.h:
* Source/WebCore/Modules/webcodecs/WebCodecsEncodedVideoChunk.h:
* Source/WebCore/Modules/webxr/WebXRJointPose.h:
* Source/WebCore/Modules/webxr/XRWebGLBinding.h:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/DOMPromiseProxy.h:
* Source/WebCore/css/CSSValue.h:
* Source/WebCore/css/DeprecatedCSSOMValue.h:
* Source/WebCore/css/StyleRule.h:
* Source/WebCore/dom/DocumentFullscreen.h:
* Source/WebCore/dom/ElementContext.h:
* Source/WebCore/dom/EventContext.h:
* Source/WebCore/html/canvas/WebGLDefaultFramebuffer.h:
* Source/WebCore/html/canvas/WebGLObject.h:
* Source/WebCore/html/canvas/WebGLUniformLocation.h:
* Source/WebCore/html/track/WebVTTParser.h:
* Source/WebCore/layout/formattingContexts/inline/InlineLine.h:
* Source/WebCore/layout/layouttree/LayoutTreeBuilder.h:
* Source/WebCore/loader/EmptyClients.cpp:
* Source/WebCore/loader/EmptyClients.h:
* Source/WebCore/page/EditorClient.cpp: Copied from Source/bmalloc/bmalloc/IsoTLSEntry.cpp.
* Source/WebCore/page/EditorClient.h:
(WebCore::EditorClient::shouldRevealCurrentSelectionAfterInsertion const): Deleted.
(WebCore::EditorClient::shouldSuppressPasswordEcho const): Deleted.
(WebCore::EditorClient::shouldRemoveDictationAlternativesAfterEditing const): Deleted.
(WebCore::EditorClient::registerAttachmentIdentifier): Deleted.
(WebCore::EditorClient::registerAttachments): Deleted.
(WebCore::EditorClient::cloneAttachmentData): Deleted.
(WebCore::EditorClient::didInsertAttachmentWithIdentifier): Deleted.
(WebCore::EditorClient::didRemoveAttachmentWithIdentifier): Deleted.
(WebCore::EditorClient::supportsClientSideAttachmentData const): Deleted.
(WebCore::EditorClient::serializedAttachmentDataForIdentifiers): Deleted.
(WebCore::EditorClient::requestCandidatesForSelection): Deleted.
(WebCore::EditorClient::handleAcceptedCandidateWithSoftSpaces): Deleted.
(WebCore::EditorClient::didDispatchInputMethodKeydown): Deleted.
(WebCore::EditorClient::supportsGlobalSelection): Deleted.
(WebCore::EditorClient::shouldAllowSingleClickToChangeSelection const): Deleted.
(WebCore::EditorClient::willChangeSelectionForAccessibility): Deleted.
(WebCore::EditorClient::didChangeSelectionForAccessibility): Deleted.
(WebCore::EditorClient::shouldDrawVisuallyContiguousBidiSelection const): Deleted.
* Source/WebCore/page/FocusController.cpp:
* Source/WebCore/page/FrameView.cpp:
* Source/WebCore/page/NavigationHistoryEntry.cpp:
* Source/WebCore/page/PageDebuggable.h:
* Source/WebCore/page/PageOverlayController.cpp:
* Source/WebCore/page/UserScript.h:
* Source/WebCore/page/linux/ResourceUsageOverlayLinux.cpp:
* Source/WebCore/svg/SVGVisitedElementTracking.h:
* Source/WebCore/workers/WorkerFontLoadRequest.h:
* Source/WebCore/workers/service/context/ServiceWorkerDebuggable.h:
* Source/WebGPU/WGSL/AST/ASTStringDumper.h:
* Source/WebKit/NetworkProcess/glib/DNSCache.h:
* Source/WebKit/Platform/IPC/MessageReceiveQueueMap.h:
* Source/WebKit/Platform/IPC/StreamConnectionEncoder.h:
* Source/WebKit/Platform/cocoa/WebPrivacyHelpers.h:
* Source/WebKit/Shared/IPCStreamTesterProxy.h:
* Source/WebKit/Shared/IPCTesterReceiver.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKInspector.mm:
* Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.h:
* Source/WebKit/UIProcess/API/gtk/WebKitPopupMenu.h:
* Source/WebKit/UIProcess/API/wpe/WebKitPopupMenu.h:
* Source/WebKit/UIProcess/Cocoa/TextCheckingController.h:
* Source/WebKit/UIProcess/DisplayLinkProcessProxyClient.h:
* Source/WebKit/UIProcess/Extensions/WebExtensionMatchPattern.h:
* Source/WebKit/UIProcess/Inspector/WebPageInspectorTargetProxy.h:
* Source/WebKit/UIProcess/Inspector/glib/RemoteInspectorHTTPServer.h:
* Source/WebKit/UIProcess/Launcher/glib/XDGDBusProxy.h:
* Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteScrollingCoordinatorProxyIOS.h:
* Source/WebKit/UIProcess/glib/DisplayVBlankMonitorDRM.h:
* Source/WebKit/UIProcess/glib/DisplayVBlankMonitorTimer.h:
* Source/WebKit/UIProcess/gtk/AcceleratedBackingStore.h:
* Source/WebKit/UIProcess/gtk/ClipboardGtk4.cpp:
* Source/WebKit/WPEPlatform/wpe/WPEBufferDMABufFormats.cpp:
* Source/WebKit/WPEPlatform/wpe/WPECursorTheme.h:
* Source/WebKit/WPEPlatform/wpe/drm/WPEDRM.h:
* Source/WebKit/WPEPlatform/wpe/wayland/WPEToplevelWayland.cpp:
(DMABufFeedback::Tranche::Tranche):
* Source/WebKit/WebProcess/EncryptedMedia/MediaKeySystemPermissionRequestManager.h:
* Source/WebKit/WebProcess/GPU/graphics/ImageBufferRemoteDisplayListBackend.cpp:
(WebKit::ImageBufferRemoteDisplayListBackend::~ImageBufferRemoteDisplayListBackend): Deleted.
* Source/WebKit/WebProcess/GPU/graphics/RemoteResourceCacheProxy.cpp:
(WebKit::RemoteResourceCacheProxy::~RemoteResourceCacheProxy): Deleted.
* Source/WebKit/WebProcess/GPU/media/RemoteAudioSourceProvider.cpp:
(WebKit::RemoteAudioSourceProvider::~RemoteAudioSourceProvider): Deleted.
* Source/WebKit/WebProcess/GPU/media/RemoteMediaPlayerMIMETypeCache.h:
* Source/WebKit/WebProcess/GPU/media/RemoteVideoCodecFactory.cpp:
(WebKit::RemoteVideoDecoderCallbacks::create):
(WebKit::RemoteVideoEncoderCallbacks::create):
(WebKit::RemoteVideoCodecFactory::~RemoteVideoCodecFactory): Deleted.
* Source/WebKit/WebProcess/GPU/webrtc/RemoteVideoFrameObjectHeapProxy.h:
(WebKit::RemoteVideoFrameObjectHeapProxy::create):
* Source/WebKit/WebProcess/GPU/webrtc/SampleBufferDisplayLayerManager.h:
* Source/WebKit/WebProcess/Gamepad/WebGamepadProvider.cpp:
(WebKit::WebGamepadProvider::WebGamepadProvider): Deleted.
(WebKit::WebGamepadProvider::~WebGamepadProvider): Deleted.
* Source/WebKit/WebProcess/InjectedBundle/API/APIInjectedBundleEditorClient.h:
(API::InjectedBundle::EditorClient::~EditorClient): Deleted.
* Source/WebKit/WebProcess/InjectedBundle/API/APIInjectedBundlePageContextMenuClient.h:
(API::InjectedBundle::PageContextMenuClient::~PageContextMenuClient): Deleted.
* Source/WebKit/WebProcess/InjectedBundle/API/APIInjectedBundlePageUIClient.h:
(API::InjectedBundle::PageUIClient::~PageUIClient): Deleted.
* Source/WebKit/WebProcess/InjectedBundle/API/c/mac/WKBundlePageBannerMac.mm:
(WebKit::PageBannerClientImpl::~PageBannerClientImpl): Deleted.
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/GObjectXPathNSResolver.cpp:
(WebKit::GObjectXPathNSResolver::~GObjectXPathNSResolver): Deleted.
* Source/WebKit/WebProcess/InjectedBundle/API/mac/WKDOMInternals.h:
(WebKit::DOMCache::DOMCache): Deleted.
* Source/WebKit/WebProcess/InjectedBundle/APIInjectedBundleFormClient.h:
(API::InjectedBundle::FormClient::~FormClient): Deleted.
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundle.cpp:
(WebKit::InjectedBundle::~InjectedBundle): Deleted.
* Source/WebKit/WebProcess/Inspector/ServiceWorkerDebuggableProxy.h:
* Source/WebKit/WebProcess/Inspector/WebInspectorBackendClient.cpp:
* Source/WebKit/WebProcess/Inspector/WebInspectorClient.cpp:
* Source/WebKit/WebProcess/MediaCache/WebMediaKeyStorageManager.h:
(WebKit::WebMediaKeyStorageManager::~WebMediaKeyStorageManager): Deleted.
* Source/WebKit/WebProcess/MediaStream/UserMediaPermissionRequestManager.h:
* Source/WebKit/WebProcess/Model/ARKitInlinePreviewModelPlayer.mm:
(WebKit::ARKitInlinePreviewModelPlayer::~ARKitInlinePreviewModelPlayer): Deleted.
* Source/WebKit/WebProcess/Network/WebResourceLoader.cpp:
(WebKit::WebResourceLoader::~WebResourceLoader): Deleted.
* Source/WebKit/WebProcess/Plugins/PDF/PDFIncrementalLoader.mm:
* Source/WebKit/WebProcess/Storage/WebServiceWorkerProvider.cpp:
(WebKit::WebServiceWorkerProvider::WebServiceWorkerProvider): Deleted.
* Source/WebKit/WebProcess/UserContent/WebUserContentController.cpp:
(WebKit::WebUserMessageHandlerDescriptorProxy::~WebUserMessageHandlerDescriptorProxy): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebCaptionPreferencesDelegate.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebCryptoClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebMediaKeySystemClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebMessagePortChannelProvider.cpp:
(WebKit::WebMessagePortChannelProvider::WebMessagePortChannelProvider): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebPasteboardOverrides.cpp:
(WebKit::WebPasteboardOverrides::WebPasteboardOverrides): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebPlatformStrategies.cpp:
(WebKit::WebPlatformStrategies::WebPlatformStrategies): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebPopupMenu.cpp:
(WebKit::WebPopupMenu::~WebPopupMenu): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebSpeechSynthesisClient.h:
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/AcceleratedSurface.cpp:
(WebKit::AcceleratedSurface::~AcceleratedSurface): Deleted.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/AcceleratedSurface.h:
* Source/WebKit/WebProcess/WebPage/FindController.cpp:
(WebKit::FindController::~FindController): Deleted.
* Source/WebKit/WebProcess/WebPage/PageBanner.h:
(WebKit::PageBanner::Client::~Client): Deleted.
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/PlatformCAAnimationRemote.h:
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/PlatformCALayerRemoteTiledBacking.cpp:
(WebKit::PlatformCALayerRemoteTiledBacking::~PlatformCALayerRemoteTiledBacking): Deleted.
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteScrollbarsController.h:
* Source/WebKit/WebProcess/WebPage/WebContextMenu.cpp:
(WebKit::WebContextMenu::~WebContextMenu): Deleted.
* Source/WebKit/WebProcess/WebPage/WebOpenPanelResultListener.cpp:
(WebKit::WebOpenPanelResultListener::~WebOpenPanelResultListener): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPageGroupProxy.cpp:
(WebKit::WebPageGroupProxy::~WebPageGroupProxy): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPageOverlay.h:
(WebKit::WebPageOverlay::Client::~Client): Deleted.
* Source/WebKit/WebProcess/WebPage/WebUndoStep.cpp:
(WebKit::WebUndoStep::~WebUndoStep): Deleted.
* Source/WebKit/WebProcess/WebPage/wc/WCBackingStore.h:
(WebKit::WCBackingStore::WCBackingStore): Deleted.
* Source/WebKit/WebProcess/cocoa/PlaybackSessionManager.mm:
(WebKit::PlaybackSessionInterfaceContext::~PlaybackSessionInterfaceContext): Deleted.
* Source/WebKit/WebProcess/cocoa/TextTrackRepresentationCocoa.h:
* Source/WebKit/WebProcess/cocoa/VideoPresentationManager.mm:
(WebKit::VideoPresentationInterfaceContext::~VideoPresentationInterfaceContext): Deleted.
* Source/WebKit/webpushd/webpushtool/WebPushToolMain.mm:
(WebKit::InjectPushMessageVerb::InjectPushMessageVerb):
(WebKit::GetPushPermissionStateVerb::GetPushPermissionStateVerb):
(WebKit::RequestPushPermissionVerb::RequestPushPermissionVerb):
* Source/WebKitLegacy/Storage/StorageTrackerClient.h:
(WebCore::StorageTrackerClient::~StorageTrackerClient): Deleted.
* Source/WebKitLegacy/Storage/WebDatabaseProvider.cpp:
(WebDatabaseProvider::WebDatabaseProvider): Deleted.
(WebDatabaseProvider::~WebDatabaseProvider): Deleted.
* Source/WebKitLegacy/WebCoreSupport/WebResourceLoadScheduler.cpp:
(WebResourceLoadScheduler::~WebResourceLoadScheduler): Deleted.
* Source/WebKitLegacy/ios/WebCoreSupport/WebFixedPositionContent.mm:
(WebFixedPositionContentData::~WebFixedPositionContentData): Deleted.
* Source/WebKitLegacy/mac/DOM/DOMCustomXPathNSResolver.mm:
(DOMCustomXPathNSResolver::~DOMCustomXPathNSResolver): Deleted.
* Source/WebKitLegacy/mac/History/BinaryPropertyList.h:
(BinaryPropertyListObjectStream::~BinaryPropertyListObjectStream): Deleted.
(BinaryPropertyListWriter::~BinaryPropertyListWriter): Deleted.
* Source/WebKitLegacy/mac/Misc/WebSharingServicePickerController.h:
(WebSharingServicePickerClient::~WebSharingServicePickerClient): Deleted.
* Source/WebKitLegacy/mac/Storage/WebStorageTrackerClient.mm:
(WebStorageTrackerClient::WebStorageTrackerClient): Deleted.
(WebStorageTrackerClient::~WebStorageTrackerClient): Deleted.
* Source/WebKitLegacy/mac/WebCoreSupport/SearchPopupMenuMac.mm:
(SearchPopupMenuMac::~SearchPopupMenuMac): Deleted.
* Source/WebKitLegacy/mac/WebCoreSupport/WebEditorClient.mm:
(WebEditorClient::~WebEditorClient): Deleted.
* Source/WebKitLegacy/mac/WebCoreSupport/WebPlatformStrategies.mm:
(WebPlatformStrategies::WebPlatformStrategies): Deleted.
* Source/WebKitLegacy/mac/WebCoreSupport/WebPluginInfoProvider.mm:
(WebPluginInfoProvider::WebPluginInfoProvider): Deleted.
(WebPluginInfoProvider::~WebPluginInfoProvider): Deleted.
* Source/bmalloc/bmalloc/Bits.h:
* Source/bmalloc/bmalloc/DeferredDecommitInlines.h:
* Source/bmalloc/bmalloc/DeferredTrigger.h:
(bmalloc::DeferredTrigger::DeferredTrigger): Deleted.
* Source/bmalloc/bmalloc/EligibilityResult.h:
(bmalloc::EligibilityResult::EligibilityResult): Deleted.
* Source/bmalloc/bmalloc/Gigacage.cpp:
* Source/bmalloc/bmalloc/IsoAllocatorInlines.h:
(bmalloc::IsoAllocator&lt;Config&gt;::~IsoAllocator): Deleted.
* Source/bmalloc/bmalloc/IsoDeallocatorInlines.h:
(bmalloc::IsoDeallocator&lt;Config&gt;::~IsoDeallocator): Deleted.
* Source/bmalloc/bmalloc/IsoDirectory.h:
(bmalloc::IsoDirectoryBaseBase::IsoDirectoryBaseBase): Deleted.
(bmalloc::IsoDirectoryBaseBase::~IsoDirectoryBaseBase): Deleted.
* Source/bmalloc/bmalloc/IsoHeapImpl.cpp:
(bmalloc::IsoHeapImplBase::~IsoHeapImplBase): Deleted.
* Source/bmalloc/bmalloc/IsoTLSAllocatorEntryInlines.h:
(bmalloc::IsoTLSAllocatorEntry&lt;Config&gt;::~IsoTLSAllocatorEntry): Deleted.
* Source/bmalloc/bmalloc/IsoTLSDeallocatorEntryInlines.h:
(bmalloc::IsoTLSDeallocatorEntry&lt;Config&gt;::~IsoTLSDeallocatorEntry): Deleted.
* Source/bmalloc/bmalloc/IsoTLSEntry.cpp:
(bmalloc::IsoTLSEntry::~IsoTLSEntry): Deleted.
* Source/bmalloc/bmalloc/IsoTLSEntry.h:
* Source/bmalloc/bmalloc/List.h:
(bmalloc::List::List): Deleted.
* Source/bmalloc/bmalloc/ProcessCheck.mm:
(bmalloc::ProcessNames::ProcessNames): Deleted.
* Source/bmalloc/bmalloc/SegmentedVector.h:
(bmalloc::SegmentedVectorIterator::~SegmentedVectorIterator): Deleted.
* Tools/DumpRenderTree/AccessibilityTextMarker.h:
(AccessibilityTextMarker::~AccessibilityTextMarker): Deleted.
(AccessibilityTextMarkerRange::~AccessibilityTextMarkerRange): Deleted.
* Tools/DumpRenderTree/WorkQueueItem.h:
(WorkQueueItem::~WorkQueueItem): Deleted.
* Tools/DumpRenderTree/cg/PixelDumpSupportCG.h:
* Tools/DumpRenderTree/ios/AccessibilityControllerIOS.mm:
(AccessibilityController::AccessibilityController): Deleted.
* Tools/DumpRenderTree/ios/AccessibilityTextMarkerIOS.mm:
(AccessibilityTextMarker::~AccessibilityTextMarker): Deleted.
(AccessibilityTextMarkerRange::~AccessibilityTextMarkerRange): Deleted.
* Tools/DumpRenderTree/mac/AccessibilityTextMarkerMac.mm:
(AccessibilityTextMarker::~AccessibilityTextMarker): Deleted.
(AccessibilityTextMarkerRange::~AccessibilityTextMarkerRange): Deleted.
* Tools/MiniBrowser/playstation/WebContext.cpp:
(WebContext::~WebContext): Deleted.
* Tools/MiniBrowser/win/BrowserWindow.h:
(BrowserWindow::~BrowserWindow): Deleted.
* Tools/TestRunnerShared/Bindings/JSWrappable.h:
(WTR::JSWrappable::~JSWrappable): Deleted.
* Tools/TestRunnerShared/UIScriptContext/UIScriptContext.h:
* Tools/TestWebKitAPI/InjectedBundleController.cpp:
(TestWebKitAPI::InjectedBundleController::InjectedBundleController): Deleted.
* Tools/TestWebKitAPI/InjectedBundleTest.h:
(TestWebKitAPI::InjectedBundleTest::~InjectedBundleTest): Deleted.
* Tools/TestWebKitAPI/Tests/IPC/IPCTestUtilities.h:
* Tools/TestWebKitAPI/Tests/JavaScriptCore/glib/TestJSC.cpp:
* Tools/TestWebKitAPI/Tests/WPEPlatform/TestView.cpp:
(TestWebKitAPI::WPEMockViewTest::~WPEMockViewTest): Deleted.
* Tools/TestWebKitAPI/Tests/WPEQt/WPEQtTest.h:
(LoadSpy::~LoadSpy): Deleted.
(LoadStartedCatcher::~LoadStartedCatcher): Deleted.
* Tools/TestWebKitAPI/Tests/WTF/CheckedRef.cpp:
* Tools/TestWebKitAPI/Tests/WTF/ListHashSet.cpp:
(TestWebKitAPI::FakeElementAnimationRareData::FakeElementAnimationRareData): Deleted.
(TestWebKitAPI::FakeElementAnimationRareData::~FakeElementAnimationRareData): Deleted.
* Tools/TestWebKitAPI/Tests/WTF/RedBlackTree.cpp:
* Tools/TestWebKitAPI/Tests/WTF/WeakPtr.cpp:
* Tools/TestWebKitAPI/Tests/WebCore/KeyedCoding.cpp:
* Tools/TestWebKitAPI/Tests/WebCore/curl/CurlMultipartHandleTests.cpp:
(TestWebKitAPI::Curl::CurlMultipartHandleTests::CurlMultipartHandleTests): Deleted.
* Tools/TestWebKitAPI/Tests/WebCore/curl/OpenSSLHelperTests.cpp:
(TestWebKitAPI::Curl::OpenSSLHelperTests::OpenSSLHelperTests): Deleted.
* Tools/TestWebKitAPI/Tests/WebKit/Geolocation.cpp:
(TestWebKitAPI::GeolocationStateTracker::~GeolocationStateTracker): Deleted.
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestAutomationSession.cpp:
* Tools/TestWebKitAPI/Tests/WebKitGLib/WebProcessTest.h:
(WebProcessTest::~WebProcessTest): Deleted.
* Tools/TestWebKitAPI/cocoa/ImageAnalysisTestingUtilities.h:
* Tools/WebKitTestRunner/InjectedBundle/AccessibilityController.cpp:
(WTR::AccessibilityController::~AccessibilityController): Deleted.
(WTR::AXThread::AXThread): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/AccessibilityTextMarker.cpp:
(WTR::AccessibilityTextMarker::~AccessibilityTextMarker): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/AccessibilityTextMarkerRange.cpp:
(WTR::AccessibilityTextMarkerRange::~AccessibilityTextMarkerRange): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/atspi/AccessibilityUIElementAtspi.cpp:
(WTR::AccessibilityUIElement::~AccessibilityUIElement): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/ios/AccessibilityUIElementIOS.mm:
(WTR::AccessibilityUIElement::~AccessibilityUIElement): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/mac/AccessibilityUIElementMac.mm:
(WTR::AccessibilityUIElement::~AccessibilityUIElement): Deleted.
* Tools/WebKitTestRunner/WorkQueueManager.cpp:
(WTR::WorkQueueItem::~WorkQueueItem): Deleted.
(WTR::WorkQueueManager::~WorkQueueManager): Deleted.
* Tools/WebKitTestRunner/gtk/EventSenderProxyGtk.cpp:
(WTR::EventSenderProxy::~EventSenderProxy): Deleted.
* Tools/WebKitTestRunner/libwpe/EventSenderProxyLibWPE.cpp:
(WTR::EventSenderProxy::~EventSenderProxy): Deleted.
* Tools/WebKitTestRunner/libwpe/PlatformWebViewClientLibWPE.cpp:
(WTR::PlatformWebViewClientLibWPE::~PlatformWebViewClientLibWPE): Deleted.
* Tools/WebKitTestRunner/win/EventSenderProxyWin.cpp:
(WTR::EventSenderProxy::~EventSenderProxy): Deleted.

Canonical link: <a href="https://commits.webkit.org/301386@main">https://commits.webkit.org/301386@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/025e493690ce8d0a71c044be5e9563df3e476a00

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/125796 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/45458 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/36209 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/132664 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/77672 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/ce6e3721-165e-40c6-830d-058d71af11ce) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/127667 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/46141 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/54017 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/95831 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/63947 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/96320823-93cf-4bfb-b023-39ad0a519410) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/128744 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/36888 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/112489 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/76324 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/1b8bb132-e742-4ace-8961-95e8ab86e04a) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/35792 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/30673 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/76134 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/117877 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/106667 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/30886 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/135343 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/124303 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/52583 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/40329 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/104298 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/53031 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/108699 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/104025 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26494 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/49394 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/27712 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/49912 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/52478 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/58287 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/157316 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/51826 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/39388 "Found 8 new JSC stress test failures: wasm.yaml/wasm/stress/wasm-js-string-builtins-returnStrictInt32.js.default-wasm, wasm.yaml/wasm/stress/wasm-js-string-builtins-returnStrictInt32.js.wasm-bbq, wasm.yaml/wasm/stress/wasm-js-string-builtins-returnStrictInt32.js.wasm-bbq-no-consts, wasm.yaml/wasm/stress/wasm-js-string-builtins-returnStrictInt32.js.wasm-collect-continuously, wasm.yaml/wasm/stress/wasm-js-string-builtins-returnStrictInt32.js.wasm-eager, wasm.yaml/wasm/stress/wasm-js-string-builtins-returnStrictInt32.js.wasm-eager-jettison, wasm.yaml/wasm/stress/wasm-js-string-builtins-returnStrictInt32.js.wasm-no-cjit, wasm.yaml/wasm/stress/wasm-js-string-builtins-returnStrictInt32.js.wasm-slow-memory (failure)") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/55175 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/53520 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WTF/wtf/MallocCommon.h:
```diff
@@ -71,7 +71,7 @@ class ForbidMallocUseForCurrentThreadScope {
 #else
     ForbidMallocUseForCurrentThreadScope() = default;
-    ~ForbidMallocUseForCurrentThreadScope() { }
+    ALWAYS_INLINE ~ForbidMallocUseForCurrentThreadScope() { }
```
It's OK to add this I guess but I don't understand why. Was code being generated? Why not just omit the destructor entirely?

cdumez replied:
For some of them, I had to keep the destructor explicit instead of defaulting it (or removing it, which would default it also). The reason is that without an explicit destructor, the compiler would start complaining about unused variables since it is able to detect that the destructor for those variables doesn't have any side effects. This is common for classes that do things only for some configurations (e.g. in debug), and then we would get warnings in other configurations (e.g. release builds).

cdumez replied:
I marked as ALWAYS_INLINE as the next best thing, to try and guarantee the compiler doesn't generate any code for these destructors that I couldn't default in order to silence compiler warnings. The ALWAYS_INLINE may or may not help, I did not validate but I figured it did not hurt.

weinig replied:
Perhaps its worth a comment (or its own macro, ALWAYS_INLINE_TO_SUPPRESS_...),s Ince its pretty non-obvious from reading the code.

cdumez replied:
To be clear, we do not need `ALWAYS_INLINE` to suppress the unused variable warning, just having a non-defaulted destructor. I added ALWAYS_INLINE only to try and make sure the compiler doesn't generate code at call sites since the destructor does nothing, which is pretty common practice in WebKit I think?

weinig replied:
Would inline ‘= default’ or no destructor generate code for destruction at call sites?

cdumez replied:
If the compiler decides to not inline, then it would generate a function call no?

darinadler commented on Source/bmalloc/bmalloc/List.h:
```diff
@@ -57,7 +57,7 @@ class List {
 
 public:
-    List() { }
+    List() = default;
```
```suggestion
```

darinadler commented on Tools/TestWebKitAPI/Tests/WPEPlatform/TestView.cpp:
```diff
@@ -44,9 +44,7 @@ class WPEMockViewTest : public WPEMockPlatformTest {
-    ~WPEMockViewTest()
-    {
-    }
+    ~WPEMockViewTest() = default;
```
Is this different from omitting the destructor entirely?

cdumez replied:
This PR was so large, it was a very mechanical regex replacement, I did not try to reason about each one to see if we would omit them entirely.

darinadler commented on Tools/TestWebKitAPI/Tests/WPEQt/WPEQtTest.h:
```diff
@@ -46,7 +46,7 @@ class LoadSpy : public QEventLoop {
     }
 
-    ~LoadSpy() { }
+    ~LoadSpy() = default;
```
```suggestion
```

darinadler commented on Tools/TestWebKitAPI/Tests/WTF/ListHashSet.cpp:
```diff
@@ -553,8 +553,8 @@ class FakeElementAnimationRareData {
 public:
-    explicit FakeElementAnimationRareData() { };
-    ~FakeElementAnimationRareData() { };
+    explicit FakeElementAnimationRareData() = default;
+    ~FakeElementAnimationRareData() = default;
```
```suggestion
```

darinadler commented on Tools/TestWebKitAPI/Tests/WebCore/curl/OpenSSLHelperTests.cpp:
```diff
@@ -37,7 +37,7 @@ namespace Curl {
 class OpenSSLHelperTests : public testing::Test {
 public:
-    OpenSSLHelperTests() { }
+    OpenSSLHelperTests() = default;
```
```suggestion
```

darinadler commented on Tools/TestWebKitAPI/Tests/WebKitGLib/TestAutomationSession.cpp:
```diff
@@ -42,9 +42,7 @@ class AutomationTest: public Test {
-    {
-    }
+    ~AutomationTest() = default;
 
```
```suggestion
```

darinadler commented on Source/WebKit/WebProcess/InjectedBundle/API/mac/WKDOMInternals.h:
```diff
@@ -59,9 +59,7 @@ namespace WebKit {
-    {
-    }
+    DOMCache() = default;
 
```
```suggestion
```

darinadler commented on Source/WebKit/WebProcess/WebPage/wc/WCBackingStore.h:
```diff
@@ -36,7 +36,7 @@ namespace WebKit {
     WTF_MAKE_TZONE_ALLOCATED(WCBackingStore);
 public:
-    WCBackingStore() { }
+    WCBackingStore() = default;
```
```suggestion
```

darinadler commented on Source/bmalloc/bmalloc/Bits.h:
```diff
@@ -379,7 +379,7 @@ class BitsImpl {
 template<size_t passedNumBits>
 class Bits : public BitsImpl<BitsWordOwner<passedNumBits>> {
 public:
-    Bits() { }
+    Bits() = default;
     
     Bits(const Bits&) = default;
     Bits& operator=(const Bits&) = default;
```
```suggestion
```

darinadler commented on Source/bmalloc/bmalloc/DeferredTrigger.h:
```diff
@@ -40,7 +40,7 @@ template<typename Config> class IsoPage;
 class DeferredTrigger {
 public:
-    DeferredTrigger() { }
+    DeferredTrigger() = default;
```
```suggestion
```

darinadler commented on Source/bmalloc/bmalloc/IsoDirectory.h:
```diff
@@ -41,8 +41,8 @@ template<typename Config> class IsoHeapImpl;
 public:
-    IsoDirectoryBaseBase() { }
-    virtual ~IsoDirectoryBaseBase() { }
+    IsoDirectoryBaseBase() = default;
```
```suggestion
```

darinadler commented on Source/bmalloc/bmalloc/SegmentedVector.h:
```diff
@@ -49,7 +49,7 @@ class SegmentedVectorIterator {
     using reference = T&;
 
-    ~SegmentedVectorIterator() { }
+    ~SegmentedVectorIterator() = default;
```
```suggestion
```

darinadler reviewed (approved):
Some of these could instead just be deleted. I made suggestions about the ones that immediately looked like that to me.

weinig commented on Source/WebKit/WebProcess/InjectedBundle/API/APIInjectedBundleEditorClient.h:
```diff
@@ -50,7 +50,7 @@ namespace InjectedBundle {
     WTF_MAKE_TZONE_ALLOCATED_INLINE(EditorClient);
 public:
-    virtual ~EditorClient() { }
+    virtual ~EditorClient() = default;
```
Not new here, but should probably be out of line to anchor the class (ala https://llvm.org/docs/CodingStandards.html#provide-a-virtual-method-anchor-for-classes-in-headers)

cdumez replied:
Done.

weinig reviewed (approved):
Do we have good rules for when to omit vs `= default` vs. empty (`Foo() { }`) is appropriate for these?

cdumez committed:
Default more constructors / destructors in WebKit
https://bugs.webkit.org/show_bug.cgi?id=300506

Reviewed by Sam Weinig and Darin Adler.

Default more constructors / destructors in WebKit.

* Source/JavaScriptCore/bytecode/Watchpoint.h:
* Source/WTF/wtf/CheckedRef.h:
* Source/WTF/wtf/ConcurrentVector.h:
* Source/WTF/wtf/MallocCommon.h:
(WTF::ForbidMallocUseForCurrentThreadScope::~ForbidMallocUseForCurrentThreadScope):
* Source/WTF/wtf/MetaAllocator.h:
* Source/WTF/wtf/PreciseSum.h:
* Source/WTF/wtf/RefTrackerMixin.h:
* Source/WTF/wtf/WeakPtrImpl.h:
* Source/WebCore/Modules/applepay/ApplePayError.h:
* Source/WebCore/Modules/gamepad/GamepadEvent.h:
* Source/WebCore/Modules/mediacontrols/MediaControlsUtils.h:
(WebCore::MediaControlsUtils::create):
* Source/WebCore/Modules/speech/SpeechRecognitionCaptureSource.h:
* Source/WebCore/Modules/streams/ReadableStreamBYOBRequest.h:
* Source/WebCore/Modules/streams/ReadableStreamReadRequest.cpp:
(WebCore::ReadableStreamDefaultReadRequest::create):
(WebCore::ReadableStreamDefaultReadIntoRequest::create):
* Source/WebCore/Modules/streams/StreamPipeToUtilities.cpp:
(WebCore::PipeToDefaultReadRequest::create):
* Source/WebCore/Modules/streams/StreamTeeUtilities.cpp:
(WebCore::TeeDefaultReadRequest::create):
(WebCore::TeeBYOBReadRequest::create):
* Source/WebCore/Modules/webcodecs/WebCodecsEncodedAudioChunk.h:
* Source/WebCore/Modules/webcodecs/WebCodecsEncodedVideoChunk.h:
* Source/WebCore/Modules/webxr/WebXRJointPose.h:
* Source/WebCore/Modules/webxr/XRWebGLBinding.h:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/DOMPromiseProxy.h:
* Source/WebCore/css/CSSValue.h:
* Source/WebCore/css/DeprecatedCSSOMValue.h:
* Source/WebCore/css/StyleRule.h:
* Source/WebCore/dom/DocumentFullscreen.h:
* Source/WebCore/dom/ElementContext.h:
* Source/WebCore/dom/EventContext.h:
* Source/WebCore/html/canvas/WebGLDefaultFramebuffer.h:
* Source/WebCore/html/canvas/WebGLObject.h:
* Source/WebCore/html/canvas/WebGLUniformLocation.h:
* Source/WebCore/html/track/WebVTTParser.h:
* Source/WebCore/layout/formattingContexts/inline/InlineLine.h:
* Source/WebCore/layout/layouttree/LayoutTreeBuilder.h:
* Source/WebCore/loader/EmptyClients.cpp:
* Source/WebCore/loader/EmptyClients.h:
* Source/WebCore/page/EditorClient.cpp: Copied from Source/bmalloc/bmalloc/IsoTLSEntry.cpp.
* Source/WebCore/page/EditorClient.h:
(WebCore::EditorClient::shouldRevealCurrentSelectionAfterInsertion const): Deleted.
(WebCore::EditorClient::shouldSuppressPasswordEcho const): Deleted.
(WebCore::EditorClient::shouldRemoveDictationAlternativesAfterEditing const): Deleted.
(WebCore::EditorClient::registerAttachmentIdentifier): Deleted.
(WebCore::EditorClient::registerAttachments): Deleted.
(WebCore::EditorClient::cloneAttachmentData): Deleted.
(WebCore::EditorClient::didInsertAttachmentWithIdentifier): Deleted.
(WebCore::EditorClient::didRemoveAttachmentWithIdentifier): Deleted.
(WebCore::EditorClient::supportsClientSideAttachmentData const): Deleted.
(WebCore::EditorClient::serializedAttachmentDataForIdentifiers): Deleted.
(WebCore::EditorClient::requestCandidatesForSelection): Deleted.
(WebCore::EditorClient::handleAcceptedCandidateWithSoftSpaces): Deleted.
(WebCore::EditorClient::didDispatchInputMethodKeydown): Deleted.
(WebCore::EditorClient::supportsGlobalSelection): Deleted.
(WebCore::EditorClient::shouldAllowSingleClickToChangeSelection const): Deleted.
(WebCore::EditorClient::willChangeSelectionForAccessibility): Deleted.
(WebCore::EditorClient::didChangeSelectionForAccessibility): Deleted.
(WebCore::EditorClient::shouldDrawVisuallyContiguousBidiSelection const): Deleted.
* Source/WebCore/page/FocusController.cpp:
* Source/WebCore/page/FrameView.cpp:
* Source/WebCore/page/NavigationHistoryEntry.cpp:
* Source/WebCore/page/PageDebuggable.h:
* Source/WebCore/page/PageOverlayController.cpp:
* Source/WebCore/page/UserScript.h:
* Source/WebCore/page/linux/ResourceUsageOverlayLinux.cpp:
* Source/WebCore/svg/SVGVisitedElementTracking.h:
* Source/WebCore/workers/WorkerFontLoadRequest.h:
* Source/WebCore/workers/service/context/ServiceWorkerDebuggable.h:
* Source/WebGPU/WGSL/AST/ASTStringDumper.h:
* Source/WebKit/NetworkProcess/glib/DNSCache.h:
* Source/WebKit/Platform/IPC/MessageReceiveQueueMap.h:
* Source/WebKit/Platform/IPC/StreamConnectionEncoder.h:
* Source/WebKit/Platform/cocoa/WebPrivacyHelpers.h:
* Source/WebKit/Shared/IPCStreamTesterProxy.h:
* Source/WebKit/Shared/IPCTesterReceiver.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKInspector.mm:
* Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.h:
* Source/WebKit/UIProcess/API/gtk/WebKitPopupMenu.h:
* Source/WebKit/UIProcess/API/wpe/WebKitPopupMenu.h:
* Source/WebKit/UIProcess/Cocoa/TextCheckingController.h:
* Source/WebKit/UIProcess/DisplayLinkProcessProxyClient.h:
* Source/WebKit/UIProcess/Extensions/WebExtensionMatchPattern.h:
* Source/WebKit/UIProcess/Inspector/WebPageInspectorTargetProxy.h:
* Source/WebKit/UIProcess/Inspector/glib/RemoteInspectorHTTPServer.h:
* Source/WebKit/UIProcess/Launcher/glib/XDGDBusProxy.h:
* Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteScrollingCoordinatorProxyIOS.h:
* Source/WebKit/UIProcess/glib/DisplayVBlankMonitorDRM.h:
* Source/WebKit/UIProcess/glib/DisplayVBlankMonitorTimer.h:
* Source/WebKit/UIProcess/gtk/AcceleratedBackingStore.h:
* Source/WebKit/UIProcess/gtk/ClipboardGtk4.cpp:
* Source/WebKit/WPEPlatform/wpe/WPEBufferDMABufFormats.cpp:
* Source/WebKit/WPEPlatform/wpe/WPECursorTheme.h:
* Source/WebKit/WPEPlatform/wpe/drm/WPEDRM.h:
* Source/WebKit/WPEPlatform/wpe/wayland/WPEToplevelWayland.cpp:
(DMABufFeedback::Tranche::Tranche):
* Source/WebKit/WebProcess/EncryptedMedia/MediaKeySystemPermissionRequestManager.h:
* Source/WebKit/WebProcess/GPU/graphics/ImageBufferRemoteDisplayListBackend.cpp:
(WebKit::ImageBufferRemoteDisplayListBackend::~ImageBufferRemoteDisplayListBackend): Deleted.
* Source/WebKit/WebProcess/GPU/graphics/RemoteResourceCacheProxy.cpp:
(WebKit::RemoteResourceCacheProxy::~RemoteResourceCacheProxy): Deleted.
* Source/WebKit/WebProcess/GPU/media/RemoteAudioSourceProvider.cpp:
(WebKit::RemoteAudioSourceProvider::~RemoteAudioSourceProvider): Deleted.
* Source/WebKit/WebProcess/GPU/media/RemoteMediaPlayerMIMETypeCache.h:
* Source/WebKit/WebProcess/GPU/media/RemoteVideoCodecFactory.cpp:
(WebKit::RemoteVideoDecoderCallbacks::create):
(WebKit::RemoteVideoEncoderCallbacks::create):
(WebKit::RemoteVideoCodecFactory::~RemoteVideoCodecFactory): Deleted.
* Source/WebKit/WebProcess/GPU/webrtc/RemoteVideoFrameObjectHeapProxy.h:
(WebKit::RemoteVideoFrameObjectHeapProxy::create):
* Source/WebKit/WebProcess/GPU/webrtc/SampleBufferDisplayLayerManager.h:
* Source/WebKit/WebProcess/Gamepad/WebGamepadProvider.cpp:
(WebKit::WebGamepadProvider::WebGamepadProvider): Deleted.
(WebKit::WebGamepadProvider::~WebGamepadProvider): Deleted.
* Source/WebKit/WebProcess/InjectedBundle/API/APIInjectedBundleEditorClient.h:
(API::InjectedBundle::EditorClient::~EditorClient): Deleted.
* Source/WebKit/WebProcess/InjectedBundle/API/APIInjectedBundlePageContextMenuClient.h:
(API::InjectedBundle::PageContextMenuClient::~PageContextMenuClient): Deleted.
* Source/WebKit/WebProcess/InjectedBundle/API/APIInjectedBundlePageUIClient.h:
(API::InjectedBundle::PageUIClient::~PageUIClient): Deleted.
* Source/WebKit/WebProcess/InjectedBundle/API/c/mac/WKBundlePageBannerMac.mm:
(WebKit::PageBannerClientImpl::~PageBannerClientImpl): Deleted.
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/GObjectXPathNSResolver.cpp:
(WebKit::GObjectXPathNSResolver::~GObjectXPathNSResolver): Deleted.
* Source/WebKit/WebProcess/InjectedBundle/API/mac/WKDOMInternals.h:
(WebKit::DOMCache::DOMCache): Deleted.
* Source/WebKit/WebProcess/InjectedBundle/APIInjectedBundleFormClient.h:
(API::InjectedBundle::FormClient::~FormClient): Deleted.
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundle.cpp:
(WebKit::InjectedBundle::~InjectedBundle): Deleted.
* Source/WebKit/WebProcess/Inspector/ServiceWorkerDebuggableProxy.h:
* Source/WebKit/WebProcess/Inspector/WebInspectorBackendClient.cpp:
* Source/WebKit/WebProcess/Inspector/WebInspectorClient.cpp:
* Source/WebKit/WebProcess/MediaCache/WebMediaKeyStorageManager.h:
(WebKit::WebMediaKeyStorageManager::~WebMediaKeyStorageManager): Deleted.
* Source/WebKit/WebProcess/MediaStream/UserMediaPermissionRequestManager.h:
* Source/WebKit/WebProcess/Model/ARKitInlinePreviewModelPlayer.mm:
(WebKit::ARKitInlinePreviewModelPlayer::~ARKitInlinePreviewModelPlayer): Deleted.
* Source/WebKit/WebProcess/Network/WebResourceLoader.cpp:
(WebKit::WebResourceLoader::~WebResourceLoader): Deleted.
* Source/WebKit/WebProcess/Plugins/PDF/PDFIncrementalLoader.mm:
* Source/WebKit/WebProcess/Storage/WebServiceWorkerProvider.cpp:
(WebKit::WebServiceWorkerProvider::WebServiceWorkerProvider): Deleted.
* Source/WebKit/WebProcess/UserContent/WebUserContentController.cpp:
(WebKit::WebUserMessageHandlerDescriptorProxy::~WebUserMessageHandlerDescriptorProxy): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebCaptionPreferencesDelegate.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebCryptoClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebMediaKeySystemClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebMessagePortChannelProvider.cpp:
(WebKit::WebMessagePortChannelProvider::WebMessagePortChannelProvider): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebPasteboardOverrides.cpp:
(WebKit::WebPasteboardOverrides::WebPasteboardOverrides): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebPlatformStrategies.cpp:
(WebKit::WebPlatformStrategies::WebPlatformStrategies): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebPopupMenu.cpp:
(WebKit::WebPopupMenu::~WebPopupMenu): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebSpeechSynthesisClient.h:
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/AcceleratedSurface.cpp:
(WebKit::AcceleratedSurface::~AcceleratedSurface): Deleted.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/AcceleratedSurface.h:
* Source/WebKit/WebProcess/WebPage/FindController.cpp:
(WebKit::FindController::~FindController): Deleted.
* Source/WebKit/WebProcess/WebPage/PageBanner.h:
(WebKit::PageBanner::Client::~Client): Deleted.
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/PlatformCAAnimationRemote.h:
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/PlatformCALayerRemoteTiledBacking.cpp:
(WebKit::PlatformCALayerRemoteTiledBacking::~PlatformCALayerRemoteTiledBacking): Deleted.
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteScrollbarsController.h:
* Source/WebKit/WebProcess/WebPage/WebContextMenu.cpp:
(WebKit::WebContextMenu::~WebContextMenu): Deleted.
* Source/WebKit/WebProcess/WebPage/WebOpenPanelResultListener.cpp:
(WebKit::WebOpenPanelResultListener::~WebOpenPanelResultListener): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPageGroupProxy.cpp:
(WebKit::WebPageGroupProxy::~WebPageGroupProxy): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPageOverlay.h:
(WebKit::WebPageOverlay::Client::~Client): Deleted.
* Source/WebKit/WebProcess/WebPage/WebUndoStep.cpp:
(WebKit::WebUndoStep::~WebUndoStep): Deleted.
* Source/WebKit/WebProcess/WebPage/wc/WCBackingStore.h:
(WebKit::WCBackingStore::WCBackingStore): Deleted.
* Source/WebKit/WebProcess/cocoa/PlaybackSessionManager.mm:
(WebKit::PlaybackSessionInterfaceContext::~PlaybackSessionInterfaceContext): Deleted.
* Source/WebKit/WebProcess/cocoa/TextTrackRepresentationCocoa.h:
* Source/WebKit/WebProcess/cocoa/VideoPresentationManager.mm:
(WebKit::VideoPresentationInterfaceContext::~VideoPresentationInterfaceContext): Deleted.
* Source/WebKit/webpushd/webpushtool/WebPushToolMain.mm:
(WebKit::InjectPushMessageVerb::InjectPushMessageVerb):
(WebKit::GetPushPermissionStateVerb::GetPushPermissionStateVerb):
(WebKit::RequestPushPermissionVerb::RequestPushPermissionVerb):
* Source/WebKitLegacy/Storage/StorageTrackerClient.h:
(WebCore::StorageTrackerClient::~StorageTrackerClient): Deleted.
* Source/WebKitLegacy/Storage/WebDatabaseProvider.cpp:
(WebDatabaseProvider::WebDatabaseProvider): Deleted.
(WebDatabaseProvider::~WebDatabaseProvider): Deleted.
* Source/WebKitLegacy/WebCoreSupport/WebResourceLoadScheduler.cpp:
(WebResourceLoadScheduler::~WebResourceLoadScheduler): Deleted.
* Source/WebKitLegacy/ios/WebCoreSupport/WebFixedPositionContent.mm:
(WebFixedPositionContentData::~WebFixedPositionContentData): Deleted.
* Source/WebKitLegacy/mac/DOM/DOMCustomXPathNSResolver.mm:
(DOMCustomXPathNSResolver::~DOMCustomXPathNSResolver): Deleted.
* Source/WebKitLegacy/mac/History/BinaryPropertyList.h:
(BinaryPropertyListObjectStream::~BinaryPropertyListObjectStream): Deleted.
(BinaryPropertyListWriter::~BinaryPropertyListWriter): Deleted.
* Source/WebKitLegacy/mac/Misc/WebSharingServicePickerController.h:
(WebSharingServicePickerClient::~WebSharingServicePickerClient): Deleted.
* Source/WebKitLegacy/mac/Storage/WebStorageTrackerClient.mm:
(WebStorageTrackerClient::WebStorageTrackerClient): Deleted.
(WebStorageTrackerClient::~WebStorageTrackerClient): Deleted.
* Source/WebKitLegacy/mac/WebCoreSupport/SearchPopupMenuMac.mm:
(SearchPopupMenuMac::~SearchPopupMenuMac): Deleted.
* Source/WebKitLegacy/mac/WebCoreSupport/WebEditorClient.mm:
(WebEditorClient::~WebEditorClient): Deleted.
* Source/WebKitLegacy/mac/WebCoreSupport/WebPlatformStrategies.mm:
(WebPlatformStrategies::WebPlatformStrategies): Deleted.
* Source/WebKitLegacy/mac/WebCoreSupport/WebPluginInfoProvider.mm:
(WebPluginInfoProvider::WebPluginInfoProvider): Deleted.
(WebPluginInfoProvider::~WebPluginInfoProvider): Deleted.
* Source/bmalloc/bmalloc/Bits.h:
* Source/bmalloc/bmalloc/DeferredDecommitInlines.h:
* Source/bmalloc/bmalloc/DeferredTrigger.h:
(bmalloc::DeferredTrigger::DeferredTrigger): Deleted.
* Source/bmalloc/bmalloc/EligibilityResult.h:
(bmalloc::EligibilityResult::EligibilityResult): Deleted.
* Source/bmalloc/bmalloc/Gigacage.cpp:
* Source/bmalloc/bmalloc/IsoAllocatorInlines.h:
(bmalloc::IsoAllocator<Config>::~IsoAllocator): Deleted.
* Source/bmalloc/bmalloc/IsoDeallocatorInlines.h:
(bmalloc::IsoDeallocator<Config>::~IsoDeallocator): Deleted.
* Source/bmalloc/bmalloc/IsoDirectory.h:
(bmalloc::IsoDirectoryBaseBase::IsoDirectoryBaseBase): Deleted.
(bmalloc::IsoDirectoryBaseBase::~IsoDirectoryBaseBase): Deleted.
* Source/bmalloc/bmalloc/IsoHeapImpl.cpp:
(bmalloc::IsoHeapImplBase::~IsoHeapImplBase): Deleted.
* Source/bmalloc/bmalloc/IsoTLSAllocatorEntryInlines.h:
(bmalloc::IsoTLSAllocatorEntry<Config>::~IsoTLSAllocatorEntry): Deleted.
* Source/bmalloc/bmalloc/IsoTLSDeallocatorEntryInlines.h:
(bmalloc::IsoTLSDeallocatorEntry<Config>::~IsoTLSDeallocatorEntry): Deleted.
* Source/bmalloc/bmalloc/IsoTLSEntry.cpp:
(bmalloc::IsoTLSEntry::~IsoTLSEntry): Deleted.
* Source/bmalloc/bmalloc/IsoTLSEntry.h:
* Source/bmalloc/bmalloc/List.h:
(bmalloc::List::List): Deleted.
* Source/bmalloc/bmalloc/ProcessCheck.mm:
(bmalloc::ProcessNames::ProcessNames): Deleted.
* Source/bmalloc/bmalloc/SegmentedVector.h:
(bmalloc::SegmentedVectorIterator::~SegmentedVectorIterator): Deleted.
* Tools/DumpRenderTree/AccessibilityTextMarker.h:
(AccessibilityTextMarker::~AccessibilityTextMarker): Deleted.
(AccessibilityTextMarkerRange::~AccessibilityTextMarkerRange): Deleted.
* Tools/DumpRenderTree/WorkQueueItem.h:
(WorkQueueItem::~WorkQueueItem): Deleted.
* Tools/DumpRenderTree/cg/PixelDumpSupportCG.h:
* Tools/DumpRenderTree/ios/AccessibilityControllerIOS.mm:
(AccessibilityController::AccessibilityController): Deleted.
* Tools/DumpRenderTree/ios/AccessibilityTextMarkerIOS.mm:
(AccessibilityTextMarker::~AccessibilityTextMarker): Deleted.
(AccessibilityTextMarkerRange::~AccessibilityTextMarkerRange): Deleted.
* Tools/DumpRenderTree/mac/AccessibilityTextMarkerMac.mm:
(AccessibilityTextMarker::~AccessibilityTextMarker): Deleted.
(AccessibilityTextMarkerRange::~AccessibilityTextMarkerRange): Deleted.
* Tools/MiniBrowser/playstation/WebContext.cpp:
(WebContext::~WebContext): Deleted.
* Tools/MiniBrowser/win/BrowserWindow.h:
(BrowserWindow::~BrowserWindow): Deleted.
* Tools/TestRunnerShared/Bindings/JSWrappable.h:
(WTR::JSWrappable::~JSWrappable): Deleted.
* Tools/TestRunnerShared/UIScriptContext/UIScriptContext.h:
* Tools/TestWebKitAPI/InjectedBundleController.cpp:
(TestWebKitAPI::InjectedBundleController::InjectedBundleController): Deleted.
* Tools/TestWebKitAPI/InjectedBundleTest.h:
(TestWebKitAPI::InjectedBundleTest::~InjectedBundleTest): Deleted.
* Tools/TestWebKitAPI/Tests/IPC/IPCTestUtilities.h:
* Tools/TestWebKitAPI/Tests/JavaScriptCore/glib/TestJSC.cpp:
* Tools/TestWebKitAPI/Tests/WPEPlatform/TestView.cpp:
(TestWebKitAPI::WPEMockViewTest::~WPEMockViewTest): Deleted.
* Tools/TestWebKitAPI/Tests/WPEQt/WPEQtTest.h:
(LoadSpy::~LoadSpy): Deleted.
(LoadStartedCatcher::~LoadStartedCatcher): Deleted.
* Tools/TestWebKitAPI/Tests/WTF/CheckedRef.cpp:
* Tools/TestWebKitAPI/Tests/WTF/ListHashSet.cpp:
(TestWebKitAPI::FakeElementAnimationRareData::FakeElementAnimationRareData): Deleted.
(TestWebKitAPI::FakeElementAnimationRareData::~FakeElementAnimationRareData): Deleted.
* Tools/TestWebKitAPI/Tests/WTF/RedBlackTree.cpp:
* Tools/TestWebKitAPI/Tests/WTF/WeakPtr.cpp:
* Tools/TestWebKitAPI/Tests/WebCore/KeyedCoding.cpp:
* Tools/TestWebKitAPI/Tests/WebCore/curl/CurlMultipartHandleTests.cpp:
(TestWebKitAPI::Curl::CurlMultipartHandleTests::CurlMultipartHandleTests): Deleted.
* Tools/TestWebKitAPI/Tests/WebCore/curl/OpenSSLHelperTests.cpp:
(TestWebKitAPI::Curl::OpenSSLHelperTests::OpenSSLHelperTests): Deleted.
* Tools/TestWebKitAPI/Tests/WebKit/Geolocation.cpp:
(TestWebKitAPI::GeolocationStateTracker::~GeolocationStateTracker): Deleted.
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestAutomationSession.cpp:
* Tools/TestWebKitAPI/Tests/WebKitGLib/WebProcessTest.h:
(WebProcessTest::~WebProcessTest): Deleted.
* Tools/TestWebKitAPI/cocoa/ImageAnalysisTestingUtilities.h:
* Tools/WebKitTestRunner/InjectedBundle/AccessibilityController.cpp:
(WTR::AccessibilityController::~AccessibilityController): Deleted.
(WTR::AXThread::AXThread): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/AccessibilityTextMarker.cpp:
(WTR::AccessibilityTextMarker::~AccessibilityTextMarker): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/AccessibilityTextMarkerRange.cpp:
(WTR::AccessibilityTextMarkerRange::~AccessibilityTextMarkerRange): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/atspi/AccessibilityUIElementAtspi.cpp:
(WTR::AccessibilityUIElement::~AccessibilityUIElement): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/ios/AccessibilityUIElementIOS.mm:
(WTR::AccessibilityUIElement::~AccessibilityUIElement): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/mac/AccessibilityUIElementMac.mm:
(WTR::AccessibilityUIElement::~AccessibilityUIElement): Deleted.
* Tools/WebKitTestRunner/WorkQueueManager.cpp:
(WTR::WorkQueueItem::~WorkQueueItem): Deleted.
(WTR::WorkQueueManager::~WorkQueueManager): Deleted.
* Tools/WebKitTestRunner/gtk/EventSenderProxyGtk.cpp:
(WTR::EventSenderProxy::~EventSenderProxy): Deleted.
* Tools/WebKitTestRunner/libwpe/EventSenderProxyLibWPE.cpp:
(WTR::EventSenderProxy::~EventSenderProxy): Deleted.
* Tools/WebKitTestRunner/libwpe/PlatformWebViewClientLibWPE.cpp:
(WTR::PlatformWebViewClientLibWPE::~PlatformWebViewClientLibWPE): Deleted.
* Tools/WebKitTestRunner/win/EventSenderProxyWin.cpp:
(WTR::EventSenderProxy::~EventSenderProxy): Deleted.

Canonical link: https://commits.webkit.org/301386@main
